### ISSUE TITLE
Borderless window still shows Status Bar and can be resized in full screen 

### ISSUE THREAD

**Issue by drmar** [godotengine/godot#37130]
**Godot version:**
3.2.1

**OS/device including version:**
macOS Catalin 10.15.3 / MacBookPro11,1

**Issue description:**
When the game runs in full screen WITHOUT the Borderless option enabled, if the cursor hits the top edge of the screen, the whole Menu Bar and Status Bar appears; also, when the cursor hits the bottom edge of the screen, the Dock appears.
![Screenshot 2020-03-18 at 10 03 43](https://user-images.githubusercontent.com/13488692/76938876-11563400-6900-11ea-9478-a6dccf4859a4.png)
![Screenshot 2020-03-18 at 10 12 57](https://user-images.githubusercontent.com/13488692/76939516-31d2be00-6901-11ea-8815-db256c4d7008.png)

**The problem:**
When the game runs in full screen WITH the Borderless option enabled, when the cursor hits the bottom edge of the screen the Dock does NOT appear anymore, but when the cursor hits the top edge of the screen, the Status Bar STILL APEEARS.
![Screenshot 2020-03-18 at 09 48 42](https://user-images.githubusercontent.com/13488692/76939674-7bbba400-6901-11ea-98f8-17047dc84d48.png)

**The second problem:**
When the game runs in Full Screen with the Borderless option enabled, but NOT with the Resizable option enabled, the window can still be resized, which looks very weird.
![Screenshot 2020-03-18 at 10 17 27](https://user-images.githubusercontent.com/13488692/76939939-02708100-6902-11ea-9438-558a8821385c.png)

**Steps to reproduce:**
Enable/Disable Full Screen, Borderless and/or Resizable options based on which problem need to be reproduced.

**Minimal reproduction project:**
Not necessary. Works with empty projects.

**Comment by akien-mga** [comment]
CC @bruvzg

**Comment by bruvzg** [comment]
> When the game runs in full screen WITH the Borderless option enabled.

These options probably should be made mutually exclusive options.

> the Status Bar STILL APEEARS.

Are you using some sort of status bar customization software? It should not appear without full menu bar.

> When the game runs in Full Screen with the Borderless option enabled, but NOT with the Resizable option enabled, the window can still be resized, which looks very weird.

Full screen option is temporary sets window to resizeable, it's macOS limitation.

**Comment by drmar** [comment]
> Are you using some sort of status bar customization software? It should not appear without full menu bar.

No, I am not using any king of Status Bar customisation software at all. In fact, the same thing happens if I have all my other software closed except Godot.

> Full screen option is temporary sets window to resizeable, it's macOS limitation.

I'm not sure I understand.
I downloaded StarCraft on the same macOS 10.15.3, because it is a game that requires you to move the cursor to the edges of the screen, and neither the Status Bar nor the Menu Bar appear when the cursor hits the top edge of the screen, nor the Dock appears when the cursor hits the bottom edge of the screen, and also the window cannot be resized while being in full screen.

**Comment by drmar** [comment]
I feel that I didn't fully explain the problem.
Download a screen recoding that betters showcases the prolem:
